src/app/parcels/ParcelsPage.tsx

```diff
@@ -130,20 +130,17 @@ const ParcelsPage: React.FC<{}> = () => {
     const theme = useTheme();
 
     useEffect(() => {
-        let staleFetch = false;
-
         (async () => {
             setIsLoading(true);
-            const formattedData = await fetchAndFormatParcelTablesData(packingDateRange);
-            if (!staleFetch) {
-                setTableData(formattedData);
+            const dateRangeToFetch = packingDateRange;
+            const tableData = await fetchAndFormatParcelTablesData(dateRangeToFetch);
+
+            if (dateRangeToFetch === packingDateRange) {
```

Comment:
If you follow the above comment, you'll see that this condition becomes false always.

`===` checks that either
- the reference to the object is the same (if the things you're comparing are reference types e.g. object)
- the value is the same (if the things you're comparing are value types e.g. number, string)

We should use explictly check that the From and To dates are identical instead - see [DayJs docs](https://day.js.org/docs/en/query/is-same)

Reply:
Thank you, makes sense